central/complianceoperator/v2/report/manager/manager_impl.go

```diff
@@ -312,11 +414,56 @@ func (m *managerImpl) handleReadyScanConfig() {
 			return
 		default:
 			if result := m.scanConfigReadyQueue.PullBlocking(m.stopper.LowLevel().GetStopRequestSignal()); result != nil {
-				log.Infof("Scan Config %s done with %d scans", result.ScanConfig.GetScanConfigName(), len(result.ScanResults))
+				log.Debugf("Scan Config %s done with %d scans and %d reports", result.ScanConfig.GetScanConfigName(), len(result.ScanResults), len(result.ReportSnapshot))
 				concurrency.WithLock(&m.watchingScanConfigsLock, func() {
 					delete(m.watchingScanConfigs, result.WatcherID)
 				})
+				go m.generateReportsFromWatcherResults(result)
 			}
 		}
 	}
 }
+
+func (m *managerImpl) generateReportsFromWatcherResults(result *watcher.ScanConfigWatcherResults) {
+	for _, snapshot := range result.ReportSnapshot {
+		if err := m.validateScanConfigResults(result); err != nil {
+			snapshot.ReportStatus.RunState = storage.ComplianceOperatorReportStatus_FAILURE
+			snapshot.GetReportStatus().CompletedAt = protocompat.TimestampNow()
+			snapshot.GetReportStatus().ErrorMsg = utils.ErrScanWatchersFailed.Error()
+			if err := m.snapshotDataStore.UpsertSnapshot(result.Ctx, snapshot); err != nil {
+				log.Errorf("Unable to upsert the snapshot %s: %v", snapshot.GetReportId(), err)
+			}
+			continue
+		}
+		snapshot.ReportStatus.RunState = storage.ComplianceOperatorReportStatus_PREPARING
+		if err := m.snapshotDataStore.UpsertSnapshot(result.Ctx, snapshot); err != nil {
+			log.Errorf("Unable to upsert the snapshot %s: %v", snapshot.GetReportId(), err)
+			continue
+		}
+		generateReportReq := &reportRequest{
+			ctx:        result.Ctx,
+			scanConfig: result.ScanConfig,
+			snapshotID: snapshot.GetReportId(),
+		}
+		if err := m.concurrencySem.Acquire(context.Background(), 1); err != nil {
+			log.Errorf("Error acquiring semaphore to run new report: %v", err)
+			continue
+		}
```

Comment:
I hope this never returns an error in normal operation, cause we will loose the reports

Reply:
That could happen if we set `RegisterIntegerSetting("ROX_REPORT_EXECUTION_MAX_CONCURRENCY", 5)` to 0... then many things would go wrong and we will error a lot. Shall we maybe add a validation to that setting that it is `>0`?

Reply:
This is left open as a candidate for a follow-up. Please resolve this comment once you make a note of it.

Reply:
Opened [ROX-26741](https://issues.redhat.com/browse/ROX-26741) but I don't think this is blocking because in a real scenario we cannot have more than 2 snapshots generating reports at the same time.